sysbuild/Kconfig.suit

```diff
@@ -90,6 +72,66 @@ config SUIT_ENVELOPE_NORDIC_TOP_CACHE_PARTITION_NUM
 
 endif # SUIT_ENVELOPE_NORDIC_TOP_IN_ROOT
 
+config SUIT_ENVELOPE_ROOT_SIGN
+	bool "Sign the target envelope"
+
+if SUIT_ENVELOPE_ROOT_SIGN
+
+choice SUIT_ENVELOPE_ROOT_SIGN_KEY_GEN
+	prompt "SUIT root envelope signing key generation"
+	default SUIT_ENVELOPE_ROOT_SIGN_KEY_GEN1
+
+	config SUIT_ENVELOPE_ROOT_SIGN_KEY_GEN1
+		bool "Key generation 1"
+
+	config SUIT_ENVELOPE_ROOT_SIGN_KEY_GEN2
+		bool "Key generation 2"
+
+	config SUIT_ENVELOPE_ROOT_SIGN_KEY_GEN3
+		bool "Key generation 3"
+endchoice
+
+config SUIT_ENVELOPE_ROOT_SIGN_KEY_ID
+	hex "The key ID used to identify the OEM root public key on the device"
+	default 0x4000AA00 if SUIT_ENVELOPE_ROOT_SIGN_KEY_GEN1
+	default 0x4000AA01 if SUIT_ENVELOPE_ROOT_SIGN_KEY_GEN2
+	default 0x4000AA02 if SUIT_ENVELOPE_ROOT_SIGN_KEY_GEN3
+	help
+	  This string is translated to the numeric KEY ID by the encryption script
+
+config SUIT_ENVELOPE_ROOT_SIGN_PRIVATE_KEY_NAME
```

Comment:
This part is quite repetitive, so you may consider adding a Kconfig template. If one day we will add support for a new signing algorithm, we will be able to to do so through a template adjustment.

Reply:
I think in general I agree but I think this would be to much work for now, as it wouldn't be obvious where to add this template file.